opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -375,19 +382,24 @@ impl BatchLogProcessor {
                             exporter.set_resource(&resource);
                         }
                         Err(RecvTimeoutError::Timeout) => {
+                            otel_debug!(
+                                name: "BatchLogProcessor.ExportingDueToTimer",
+                            );
                             let _ = export_with_timeout_sync(
-                                remaining_time,
+                                config.max_export_timeout,
                                 &mut exporter,
                                 logs.split_off(0),
                                 &mut last_export_time,
                             );
                         }
-                        Err(err) => {
-                            // TODO: this should not happen! Log the error and continue for now.
-                            otel_error!(
-                                name: "BatchLogProcessor.InternalError",
-                                error = format!("{}", err)
+                        Err(RecvTimeoutError::Disconnected) => {
+                            // Channel disconnected, only thing to do is break
+                            // out (i.e exit the thread)
+                            otel_debug!(
+                                name: "BatchLogProcessor.ThreadExiting",
+                                reason = "MessageReceiverDisconnected"
```

Comment:
The reason for this error is Sender(s) disconnected and not Receiver disconnected.

Reply:
fixed